pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -85,11 +90,35 @@ type event struct {
 	oldObj interface{}
 }
 
+type concurrentUIDToNode struct {
+	*sync.RWMutex
+	uidToNode map[types.UID]*node
+}
+
+func (m *concurrentUIDToNode) Write(uid types.UID, node *node) {
```

Comment:
Done, but I let the Read keep using uid as a parameter, because we don't have a node when handling Events.